3-projects/modules/single_project/main.tf

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  host_network = data.google_compute_network.shared_vpc
+}
+
+module "project" {
+  source                      = "terraform-google-modules/project-factory/google"
+  version                     = "~> 7.0"
+  random_project_id           = "true"
+  impersonate_service_account = var.impersonate_service_account
+  default_service_account     = "depriviledge"
+  activate_apis               = var.activate_apis
+  name                        = "${var.project_prefix}-${var.environment}"
+  org_id                      = var.org_id
+  billing_account             = var.billing_account
+  folder_id                   = var.folder_id
+
+  shared_vpc         = local.host_network.project
+  shared_vpc_subnets = local.host_network.subnetworks_self_links
+
+  labels = {
+    environment      = var.environment
+    cost_centre      = var.cost_centre
+    application_name = var.application_name
+  }
+}
+
+/******************************************
+  Project subnets
+ *****************************************/
+module "networking_project" {
+  source = "../../modules/project_subnet"
+
+  project_id       = module.project.project_id
+  application_name = var.application_name
+
+  enable_networking   = var.enable_networking
+  vpc_host_project_id = local.host_network.project
+  vpc_self_link       = local.host_network.self_link
+  ip_cidr_range       = var.subnet_ip_cidr_range
+  secondary_ranges    = var.subnet_secondary_ranges
+}
+
+module "dns_prod" {
```

Comment:
can we change the resource name here. I think it probably worth making this optional as well as the comment above describes. We can do the same for the variables, so its easy for someone to enable if they want it, otherwise they can discard.

Reply:
Would still be good to rename this resource to something just like DNS as the single_project module is generic.